ext/fs/interface.rs

```diff
@@ -86,26 +88,24 @@ pub trait AccessCheckFn:
   for<'a> FnMut(
   Cow<'a, Path>,
   &'a OpenOptions,
-  &'a dyn crate::GetPath,
-) -> FsResult<CheckedPath<'a>>
+) -> Result<CheckedPath<'a>, PermissionCheckError>
 {
 }
+
 impl<T> AccessCheckFn for T where
   T: for<'a> FnMut(
     Cow<'a, Path>,
     &'a OpenOptions,
-    &'a dyn crate::GetPath,
-  ) -> FsResult<CheckedPath<'a>>
+  ) -> Result<CheckedPath<'a>, PermissionCheckError>
 {
 }
```

Comment:
I'll open a follow-up PR that removes this and requires passing `CheckedPath` to `FileSystem`